packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -934,7 +938,7 @@ export class AdapterClass extends EventEmitter {
         /**
          * Promise-version of `Adapter.checkPassword`
          */
-        this.checkPasswordAsync = tools.promisifyNoError(this.checkPassword, this);
+        this.checkPasswordAsync = tools.promisifyNoError(this.checkPassword, this, ['success', 'user']);
```

Comment:
Yes there is a typing issue, but what you do here is a breaking change because before the promisified method returned the same array as the non-promisified one but with this change this gets adjusted to an object with two keys which is a breaking change. Is this really intended?

This would lead to the start of controller 8.0 if it is really needed. else the typing fix is that it returns [boolean, string]

Reply:
PS: The broken tests show exactly this breaking nature.